### PR TITLE
Chore: Ensure events for state changes [SC-406]

### PR DESCRIPTION
**Pull request by fabianschu**
DONE:
* added a bunch of events and internal setters
* added back metadata as param for ModuleInitialized cause I realized it's needed to be able to compute the module id

DID NOT DO:
* ProtocolFeeCollected event for buy and sell orders (to save gas for end users)
* no events for modifications on `_bitsPerToken` and `_accountBits` on Elastic Receipt Token

### CODE REVIEW

**Comment by marvinkruse** [comment]
A general thing I noticed is that some of the internal functions you added had an underscore in their naming (i.e. `_setDefaultProtocolTreasury`) and some didn't (i.e. `getModuleFunctionHash`) - was there are reason to this?

Also the variables are named sometimes with and without underscores, like these examples:
```Solidity
function _setDefaultProtocolTreasury(address _defaultProtocolTreasury) internal validAddress(_defaultProtocolTreasury) {
   // ...
}

function _setWorkflowTreasury(address workflow, address treasury) internal validAddress(treasury) {
   // ...
}
```

I don't wanna say that this is on you, as we don't have proper guidelines in place yet and we all probably just didn't do a good job so far ourselves, so this might be completely on us. But let's aim to no introduce any more unnecessary inconsistencies at least 😁

Could you double check and make sure of that? That would be amazing 🙏

**Comment by fabianschu** [comment]
Where I added internal setters I prefixed them with an underscore. As for the parameter names, I just copied them from the external functions (didn't rename anything). This seemed the the most consistent approach to me that doesn't involve cleaning up existing inconsistencies. The examples you are stating stem from these existing inconsistencies. This is how `dev` looks like currently for example :
Screenshot 2024-07-17 at 11 02 32


Now I agree that this annoying and I could clean this up. The question is though, how clean do we want to go. I could either
1. clean up the functions that I touched
2. clean up the codebase

If we wanna go with 2. I suggest to do it in a new PR. What do you think? Anyways, looking at the codebase it is not clear to me where and how we want to use underscores, or phrased differently which convention we want to follow more generally. So if I clean up stuff, which conventions should I follow? @marvinkruse

**Comment by marvinkruse** [comment]
Yeah that makes sense @fabianschu - then let's just not do this here, and just add your changes so they conform to whatever you find the contracts (which you did), otherwise it just gets too messy. So all good 🙏

**Comment by marvinkruse** [comment]
I merged `dev` into this one as there were quite a few changes because of audit fix merges from Macro. You can see that in [this commit in this PR](https://github.com/InverterNetwork/contracts/pull/612/commits/fa44bc2a5f1d4178d7501556a15a7e3adb93f2cd) the `MinimumStakeSet` was removed again, which is because the whole queue concept of the `KPIRewarder` has been removed, and thus there is no concept of a minimum stake anymore. 

I then added another commit [here](https://github.com/InverterNetwork/contracts/pull/612/commits/4beeb617e430205a5ef45e7dd18028a849a125e3), to remove this event in the first place, otherwise it wouldn't be used.

Does that make sense? Hope that is okay with you?

File: src/modules/fundingManager/bondingCurve/FM_BC_Bancor_Redeeming_VirtualSupply_v1.sol [hunk 1]
```diff
@@ -181,7 +181,7 @@ contract FM_BC_Bancor_Redeeming_VirtualSupply_v1 is
         // Set selling functionality to open if true. By default selling is false
         sellIsOpen = bondingCurveProperties.sellIsOpen;
 
-        emit CollateralTokenSet(_acceptedToken, collateralTokenDecimals);
+        emit ReserveTokenSet(_acceptedToken, collateralTokenDecimals);
```
Comment (marvinkruse) [inline src/modules/fundingManager/bondingCurve/FM_BC_Bancor_Redeeming_VirtualSupply_v1.sol]: Is this naming something we all agree on? I know that we call it collateral token everywhere else, so this looks a bit "off" to me.

File: src/modules/logicModule/LM_PC_KPIRewarder_v1.sol [hunk 2]
```diff
@@ -187,6 +187,7 @@ contract LM_PC_KPIRewarder_v1 is
 
         assertionPending = true;
 
+        // TODO
```
Comment (marvinkruse) [inline src/modules/logicModule/LM_PC_KPIRewarder_v1.sol]: Left a `TODO` here

**Review by marvinkruse — APPROVED** [review]

**Review by marvinkruse — DISMISSED** [review]
Sorry, didn't mean to approve already, just left these two comments, nothing big

File: src/modules/fundingManager/bondingCurve/FM_BC_Bancor_Redeeming_VirtualSupply_v1.sol (hunk 1 above)
Comment (fabianschu) [reply src/modules/fundingManager/bondingCurve/FM_BC_Bancor_Redeeming_VirtualSupply_v1.sol]: Yea, I was not sure about the naming. I just wanted to use the same event for both BondingCurve and Rebasing_FM when it comes to the token that both modules. My understanding is that the token that both modules take can be considered a reserve token, I wasn't so sure if the token can also be called a collateral token for the rebasing fm, but if so I'd also prefer to call it CollateralToken.

File: src/external/fees/FeeManager_v1.sol [hunk 3]
```diff
@@ -265,30 +266,90 @@ contract FeeManager_v1 is
...
     /// @inheritdoc IFeeManager_v1
     function setDefaultIssuanceFee(uint _defaultIssuanceFee)
         external
         onlyOwner
+    {
+        _setDefaultIssuanceFee(_defaultIssuanceFee);
+    }
+
+    /// @inheritdoc IFeeManager_v1
+    function setCollateralWorkflowFee(
```
Comment (FHieser) [inline src/external/fees/FeeManager_v1.sol]: Why exactly are you adding an internal function here?

File: src/factories/OrchestratorFactory_v1.sol [hunk 4]
```diff
@@ -133,9 +133,9 @@ contract OrchestratorFactory_v1 is
         ) {
             revert OrchestratorFactory__InvalidBeacon();
         }
-
         beacon = beacon_;
         moduleFactory = moduleFactory_;
+        emit OrchestratorFactoryInitialized(address(beacon_), moduleFactory_);
```
Comment (FHieser) [inline src/factories/OrchestratorFactory_v1.sol:138]: Where test?

File: src/factories/ModuleFactory_v1.sol [hunk 5]
```diff
@@ -147,6 +147,8 @@ contract ModuleFactory_v1 is
                 initialMetadataRegistration[i], initialBeaconRegistration[i]
             );
         }
+
+        emit GovernorSet(_governor);
```
Comment (FHieser) [inline src/factories/ModuleFactory_v1.sol:151]: Where test?

File: src/modules/base/IModule_v1.sol [hunk 6]
```diff
@@ -19,9 +19,9 @@ interface IModule_v1 {
 
     /// @notice Module has been initialized.
     /// @param parentOrchestrator The address of the orchestrator the module is linked to.
-    /// @param moduleTitle The title of the module.
+    /// @param metadata The metadata of the module.
     event ModuleInitialized(
-        address indexed parentOrchestrator, string moduleTitle
+        address indexed parentOrchestrator, Metadata metadata
```
Comment (FHieser) [inline src/modules/base/IModule_v1.sol:24]: Structs cant really be represented in the events as far as I know.
If you want to emit the whole metadata struct you have to extend the event to contain every field of it

File: src/modules/fundingManager/IFundingManager_v1.sol [hunk 7]
```diff
@@ -37,6 +37,10 @@ interface IFundingManager_v1 {
     /// @param _amount The amount of underlying tokens transfered.
     event TransferOrchestratorToken(address indexed _to, uint _amount);
 
+    /// @notice Event emitted when collateral token has been set
+    /// @param token The token that serves as collateral token making up the curve's reserve
```
Comment (FHieser) [inline src/modules/fundingManager/IFundingManager_v1.sol:41]: Not every fundingManager is made up of a curve. With that the comment here is incorrect

Comment (FHieser) [reply src/modules/fundingManager/IFundingManager_v1.sol:41]: In general I dont know if I would use the word collateral here.
"Collateral is an item of value pledged to secure a loan." Is the first sentence on google that pops up for the word and I dont think thats the case for a rebasing or Deposivault FundingManager.
Maybe OrchestratorToken? (Dont know about that one though)

File: src/modules/logicModule/LM_PC_RecurringPayments_v1.sol [hunk 8]
```diff
@@ -116,12 +116,15 @@ contract LM_PC_RecurringPayments_v1 is
...
-        epochLength = abi.decode(configData, (uint));
+        uint newEpochLength = abi.decode(configData, (uint));
+        epochLength = newEpochLength;
 
         // revert if not at least 1 week and at most a year
         if (epochLength < 1 weeks || epochLength > 52 weeks) {
             revert Module__LM_PC_RecurringPayments__InvalidEpochLength();
         }
+
+        emit EpochLengthSet(newEpochLength);
```
Comment (FHieser) [inline src/modules/logicModule/LM_PC_RecurringPayments_v1.sol:127]: Where test?

File: src/modules/logicModule/LM_PC_Staking_v1.sol [hunk 9]
```diff
@@ -239,16 +239,21 @@ contract LM_PC_Staking_v1 is
...
 
         // If trigger address is 0 then its not a user
+        uint earned;
         if (triggerAddress != address(0)) {
-            rewards[triggerAddress] = _earned(triggerAddress, rewardValue);
+            earned = _earned(triggerAddress, rewardValue);
+            rewards[triggerAddress] = earned;
             userRewardValue[triggerAddress] = rewardValue;
         }
+        emit Updated(triggerAddress, newRewardValue, newLastUpdate, earned);
```
Comment (FHieser) [inline src/modules/logicModule/LM_PC_Staking_v1.sol:256]: Where test?

File: src/modules/logicModule/LM_PC_Staking_v1.sol [hunk 10]
```diff
@@ -262,9 +267,11 @@ contract LM_PC_Staking_v1 is
             return rewardValue;
         }
 
-        return (_getRewardDistributionTimestamp() - lastUpdate) // Get the time difference between the last time it was updated and now (or in case the reward period ended the rewardEnd timestamp)
```
Comment (FHieser) [inline src/modules/logicModule/LM_PC_Staking_v1.sol]: I explicitly structured the comments here in a way to explain this complex interaction in a easier way.
You can set brackets here, but please keep it in the original structure

File: src/modules/logicModule/LM_PC_Staking_v1.sol [hunk 11]
```diff
@@ -354,5 +361,6 @@ contract LM_PC_Staking_v1 is
             revert Module__LM_PC_Staking_v1__InvalidStakingToken();
         }
         stakingToken = _token;
+        emit StakingTokenSet(_token);
```
Comment (FHieser) [inline src/modules/logicModule/LM_PC_Staking_v1.sol:362]: Where test

File: src/modules/paymentProcessor/PP_Streaming_v1.sol [hunk 12]
```diff
@@ -625,6 +625,8 @@ contract PP_Streaming_v1 is Module_v1, IPP_Streaming_v1 {
 
         // pop the last element of the array
         activePaymentReceivers[client].pop();
+
+        emit PaymentReceiverRemoved(client, paymentReceiver);
```
Comment (FHieser) [inline src/modules/paymentProcessor/PP_Streaming_v1.sol:629]: test

**Review by FHieser — CHANGES_REQUESTED** [review]
See comments

File: src/modules/base/IModule_v1.sol (hunk 6 above)
Comment (fabianschu) [reply src/modules/base/IModule_v1.sol:24]: No, they are emitted correctly, I'm using this event in the indexer :P

File: src/external/fees/FeeManager_v1.sol (hunk 3 above)
Comment (fabianschu) [reply src/external/fees/FeeManager_v1.sol]: You're right, this one is redundant, because it is not used in the `init`. I'll remove it

**Comment by Zitzak** [comment]
Given we talked about the Omega audit yesterday, and the need for them to audit additional contracts, do you think some of these code changes should be included? @FHieser

**Comment by FHieser** [comment]
> Given we talked about the Omega audit yesterday, and the need for them to audit additional contracts, do you think some of these code changes should be included? @FHieser

@Zitzak  You mean in the audit itself or that this PR is needed before the launch?
In the first case no, I think the changes here as so small that they dont need to be audited
In the second case, comes down to @fabianschu here how much this is needed from the frontend

**Comment by Zitzak** [comment]
Yeah I meant the audit itself, thank you for answering this. In the case of the code being needed for launch I would say that is a yes, as we need all those events etc for the indexer
@FHieser

**Review by FHieser — APPROVED** [review]

**Comment by marvinkruse** [comment]
Personally felt like `CollateralToken` was better then `OrchestratorToken`, does everyone else think so or do all of you like the "new" way of calling it `OrchestratorToken`? I was happy to see that old naming go and be replaced with `CollateralToken` when we did that back then :D @0xNuggan @FHieser @Zitzak  

**Edit**: Resolved, see blow.

**Review by marvinkruse — COMMENTED** [review]
Besides my comment above, it looks great!

**Comment by FHieser** [comment]
Lol,
when Fabi and I talked about it we joked that this would come up xD
The problem I raised is that Collateral Token does work in the context of issuance, but not in the direction of  general workflow token.
The token has a double role here and I think collateral Token doesnt fit here. I fully understand that OrchestratorToken kinda feels bad, but in my opinion at last it fits better as a "general" naming of the token.
I think we are all open to a better naming, if you have any good idea im happy too if we take that over.
@marvinkruse

**Comment by marvinkruse** [comment]
> Lol, when Fabi and I talked about it we joked that this would come up xD 

Haha, I was fearing this 😂 

> The problem I raised is that Collateral Token does work in the context of issuance, but not in the direction of general workflow token. The token has a double role here and I think collateral Token doesnt fit here. I fully understand that OrchestratorToken kinda feels bad, but in my opinion at last it fits better as a "general" naming of the token. I think we are all open to a better naming, if you have any good idea im happy too if we take that over. @marvinkruse

Makes sense, yeah. The `OrchestratorToken` name is crap, but I can't come up with a better naming right now, so let's just keep it until we have a better way. I'll bring this up again in a different place. It's fine here!

**Review by marvinkruse — APPROVED** [review]
looks good, ser!

**Pull request merged**